### ISSUE TITLE
"invoke" for functions

### ISSUE THREAD

**Issue by mattheww** [rust-lang/fls#462]
The spec relies on the notion of "invoking" a function in several places, but the process is never defined.

For example §6.12.1:9 (`fls_yrr1s0tucgvh`) says:

> If the callee type is a function item type or a function pointer type, then the value is the result of invoking the corresponding function with the argument operands.

Compare §6.19:7 (`fls_bqmwlona6l5w`) for return expression, which is written in terms of activation frames and transfer of control.

Something needs to say that a pattern match happens between the argument operands and the patterns represented by the function's `FunctionParameterPattern`s, effectively resulting in additional local variables.

Compare this text from the Rust Reference:

> The block of a function is conceptually wrapped in a block that binds the argument patterns and then returns the value of the function's block.

This needs to tie up with §14.4.1:3 "The binding of a function parameter is in scope within the related function body", and with §15.5 "Passing Conventions".

A different definition would be needed for external functions (and that one has to deal with variadics).

**Comment by PLeVasseur** [comment]
Hi @mattheww -- it's been a while since this was opened. Do you feel it's still relevant?

**Comment by mattheww** [comment]
Yes, I think so. I've checked the later git history for references to "invoke" or "invoking" and I don't see any sign of a definition appearing.
